auth0_flutter/lib/auth0_flutter_web.dart

```diff
@@ -0,0 +1,25 @@
+import 'package:auth0_flutter_platform_interface/auth0_flutter_platform_interface.dart';
+import 'src/version.dart';
+
+class Auth0Web {
+  final Account _account;
+
+  final UserAgent _userAgent =
+      UserAgent(name: 'auth0-flutter', version: version);
+
+  Auth0Web(final String domain, final String clientId)
+      : _account = Account(domain, clientId);
+
+  Future<Credentials?> onLoad() async {
+    await Auth0FlutterWebPlatform.instance.initialize(_account);
+
+    // Get credentials here and return them on load?
```

Comment:
Could the consumer use the result to check if the user is logged in?

Reply:
If so, I'd say go for it.

Reply:
Yes, I suppose the idea would be if the returned `Credentials` is not null, the user could be considered logged in. The use case I imagined would be if you were to initialize it in `initState`, the developer could then set up their initial app state accordingly:

```dart
@override
void initState() {
  super.initState();

  auth0Web =
      Auth0Web(dotenv.env['AUTH0_DOMAIN']!, dotenv.env['AUTH0_CLIENT_ID']!);

  // call to initialize the client and auto-handle callback or check session
  auth0Web.onLoad().then((creds) {
    if (creds != null) {
      // set "logged in" state here
    }
  });
}
```